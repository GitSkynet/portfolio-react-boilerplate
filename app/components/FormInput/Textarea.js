import Input from './Input';
import typography, { baseFontRegular } from '../../styles/templates/typography';

export const Textarea = Input.withComponent('textarea').extend`
  ${baseFontRegular}
  ${typography.bodyCopy}
  appearance: none;
  height: 2em;
  overflow: hidden;
  resize: none;
`;

export default Textarea;
