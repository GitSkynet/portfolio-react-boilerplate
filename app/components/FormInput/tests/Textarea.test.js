import React from 'react';
import { shallow } from 'enzyme';

import Textarea from '../Textarea';

describe('<Textarea />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<Textarea />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
