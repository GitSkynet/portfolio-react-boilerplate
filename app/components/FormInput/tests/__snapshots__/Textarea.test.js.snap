// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> should render and match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <textarea
    className="sc-bwzfXH eYKWwB"
/>,
  "nodes": Array [
    <textarea
      className="sc-bwzfXH eYKWwB"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <styled.textarea />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "attrs": Object {},
        "context": Object {
          "__styled-components-stylesheet__": undefined,
          "__styled-components__": undefined,
        },
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "generatedClassName": "eYKWwB",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <textarea
          className="sc-bwzfXH eYKWwB"
/>,
        "_debugID": 2,
        "_renderedOutput": <textarea
          className="sc-bwzfXH eYKWwB"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <styled.textarea />,
}
`;
